### PR TITLE
Use mullvad-version for deb version

### PR DESCRIPTION
**Pull request by raksooo**
This PR switches the deb version to be generated using `mullvad-version`.

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051)

### CODE REVIEW

**Review by faern — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @raksooo)

___
*[`gui/tasks/distribution.js` line 7 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051#-NEXQjh32UZu2jR_lCyh:-NEXQjh32UZu2jR_lCyi:b-ra0xjj) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/169121cd2a078fd89abb01672464658c47fe48f5/gui/tasks/distribution.js#L7)):*
> ```JavaScript
> const parseSemver = require('semver/functions/parse');
> const { notarize } = require('electron-notarize');
> const { version } = require('../package.json');
> ```

:partying_face: 
___
*[`gui/tasks/distribution.js` line 418 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051#-NEXQm8v9l1b6pODAC0y:-NEXQm8v9l1b6pODAC0z:b-z1789a) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/38188086451807c37ea8869a84c79123f0d12124/gui/tasks/distribution.js#L418)):*
> ```JavaScript
>   let [version, ...prereleaseParts] = productVersion([]).split('-');
>   let [major, minor] = version.split('.');
>   let prerelease = prereleaseParts.join('-');
> ```

Can't we use the `limit` argument to `split` above to not have to later join it back together again?

**Review by faern — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @raksooo)

___
*[`gui/tasks/distribution.js` line 423 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051#-NEXR2jrAqNnyE4t8A2Q:-NEXR2jrAqNnyE4t8A2R:b-vwe4q8) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/38188086451807c37ea8869a84c79123f0d12124/gui/tasks/distribution.js#L423)):*
> ```JavaScript
>       return `${major}.${minor}~${prerelease}`;
>     }
>     return `${major}.${minor}-${prerelease}`;
> ```

Not strictly related to what this PR tries to fix. But we should consider if we can format the `dev-afafaf` part in some way so semver understands it better. Since currently it's backwards that dev versions come *after* stable versions. As mentioned here: https://github.com/killercup/cargo-edit/issues/782#issuecomment-1242209495

**Review by faern — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @raksooo)

___
*[`gui/tasks/distribution.js` line 423 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051#-NEXR2jrAqNnyE4t8A2Q:-NEXRfP-0PxO7WmCSMGL:b96d12u) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/38188086451807c37ea8869a84c79123f0d12124/gui/tasks/distribution.js#L423)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Not strictly related to what this PR tries to fix. But we should consider if we can format the `dev-afafaf` part in some way so semver understands it better. Since currently it's backwards that dev versions come *after* stable versions. As mentioned here: https://github.com/killercup/cargo-edit/issues/782#issuecomment-1242209495
</blockquote></details>

Basically changing `-dev-afafaf` into `+dev-afafaf`

**Review by faern — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[`gui/tasks/distribution.js` line 418 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051#-NEXQm8v9l1b6pODAC0y:-NE_JPIKC8FnYxWPcdmN:b-zbtgsd) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/38188086451807c37ea8869a84c79123f0d12124/gui/tasks/distribution.js#L418)):*
<details><summary><i>Previously, raksooo (Oskar Nyberg) wrote…</i></summary><blockquote>

The problem with `limit` is that it ignores the rest of the items. We would like it to split it into `['2022.5', 'beta1-dev-aaaaaa']` but it splits to `['2022.5', 'beta1']`.

The alternative would be to do something like this
```
const version = productVersion([]);
const separatorIndex = version.indexOf('-');
const [major, minor] = version.substring(0, separatorIndex).split('.');
const prerelease = version.substring(separatorIndex);
```
Maybe that's easier to read?
</blockquote></details>

Ok, too bad! Well, then I think what we currently have is about as readable as that suggestion. Let's go with what we have IMO.
___
*[`gui/tasks/distribution.js` line 423 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/4051#-NEXR2jrAqNnyE4t8A2Q:-NE_Je3lE5WPVXvhRPkT:b73c362) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/38188086451807c37ea8869a84c79123f0d12124/gui/tasks/distribution.js#L423)):*
<details><summary><i>Previously, raksooo (Oskar Nyberg) wrote…</i></summary><blockquote>

Okay, then we'll have to remember to change this as well.
</blockquote></details>

The fact that we call it "prerelease parts" indicates we are doing it wrong. Because `-dev` is not a prerelease tag to us :see_no_evil:
